### PR TITLE
Allow task to be in fact an array of tasks

### PR DESCRIPTION
**Pull request by wkwiatek**
It allows to spawn a process that could have some other tasks which run in synchronous way.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you provide an example of how it would be used?

**Comment by wkwiatek** [comment]
Sure. We have something like this: 
```
tasks: [
  ...
  'bower_concat',
  ['sprite', 'less'],
  'handlebars',
  ...
]
```
And now `less` starts after `sprite` finishes while the rest is running concurrently.

I'm aware that if you'd like to include such a feature there is a need to update docs and probably create some tests regarding that but it was faster to share the code itself first.

File: tasks/concurrent.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ module.exports = function (grunt) {
 		async.eachLimit(tasks, opts.limit, function (task, next) {
 			var cp = grunt.util.spawn({
 				grunt: true,
-				args: [task].concat(flags),
+				args: task instanceof Array ? task.concat(flags) : [task].concat(flags),
```
Comment (sindresorhus) [inline tasks/concurrent.js]:
```js
args: arrify(task).concat(flags)
```

https://github.com/sindresorhus/arrify

**Comment by sindresorhus** [comment]
> I'm aware that if you'd like to include such a feature there is a need to update docs and probably create some tests regarding that but it was faster to share the code itself first.

I'm ok with adding this if you're willing to put in the work for the above :)

**Comment by wkwiatek** [comment]
@sindresorhus I finally found some time to finish that. Could you take a look?

File: test/test.js [hunk 2]
```diff
@@ -13,6 +13,21 @@ describe('concurrent', function () {
 		assert(pathExists.sync(path.join(__dirname, 'tmp/3')));
 	});
 
+	it('runs grunt task sequence successfully', function (done) {
+		fs.stat(path.join(__dirname, 'tmp/5'), function (err, file5) {
+			if (err) {
+				done();
```
Comment (sindresorhus) [inline test/test.js]: you need to `return` below here.

File: test/test.js [hunk 3]
```diff
@@ -13,6 +13,21 @@ describe('concurrent', function () {
...
+			}
+			fs.stat(path.join(__dirname, 'tmp/6'), function (err, file6) {
+				if (err) {
+					done();
```
Comment (sindresorhus) [inline test/test.js]: you need to `return` below here.

File: test/test.js [hunk 4]
```diff
@@ -13,6 +13,21 @@ describe('concurrent', function () {
 		assert(pathExists.sync(path.join(__dirname, 'tmp/3')));
 	});
 
+	it('runs grunt task sequence successfully', function (done) {
+		fs.stat(path.join(__dirname, 'tmp/5'), function (err, file5) {
+			if (err) {
+				done();
+			}
+			fs.stat(path.join(__dirname, 'tmp/6'), function (err, file6) {
```
Comment (sindresorhus) [inline test/test.js]: Just use `fs.statSync`. Above too.

File: Gruntfile.js [hunk 5]
```diff
@@ -65,6 +66,22 @@ module.exports = function (grunt) {
 		grunt.file.write('test/tmp/3');
 	});
 
+	grunt.registerTask('test4', function () {
+		console.log('test4');
+		grunt.file.write('test/tmp/4');
+	});
+
+	grunt.registerTask('test5', function () {
```
Comment (sindresorhus) [inline Gruntfile.js]: I think you can do `var done = this.async()` and do a setTimeout instead of the sleep hack.

Comment (wkwiatek) [inline Gruntfile.js]: The problem is I didn't want this task to be asynchronous. It hack simply blocks any other tasks for 1s so I can be sure that these are made one after another.

File: test/test.js (hunk 4 above)
Comment (wkwiatek) [inline test/test.js]: Ok - thanks! Will do.

**Pull request closed without merging**